### PR TITLE
Repertoire frontend

### PR DESCRIPTION
**Pull request by BohdanOne**

### CODE REVIEW

File: client/src/components/RepertoirePage.jsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      day: null,
+      movies: [],
+      shows: []
+    }
+  }
+
+  componentDidMount = async () => {
+    await this.getMovieList();
+    await this.checkDate();
+    await this.getShowList();
```
Comment (DominikKrajcer) [inline client/src/components/RepertoirePage.jsx]: You should utilize a store (Redux) to invoke actions of that purpose. Create 2 action creators getMovies and getShows (try to simplify variables name, but keep readability) which could return two action with fetched data and so on.

Comment (DominikKrajcer) [reply client/src/components/RepertoirePage.jsx]: Btw, why have you used **await this.checkDate()** even if it's not async function?

File: client/src/components/RepertoirePage.jsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  render() {
+    return (
+      <RepertoirePageContainer>
+        <Header>On screen this week</Header>
+        {this.state.shows.length > 0 ?
+          this.state.shows.map(show => <Show
+            show={ show }
+            key={ show._id }
+            movie={ this.state.movies.find(movie => movie._id === show.movie)}
+          />):
```
Comment (DominikKrajcer) [inline client/src/components/RepertoirePage.jsx]: That logic should be moved to i.e. renderShow.

File: client/src/components/RepertoirePage.jsx (hunk 1 above)
Comment (BohdanOne) [reply client/src/components/RepertoirePage.jsx]: > You should utilize a store (Redux) to invoke actions of that purpose. Create 2 action creators getMovies and getShows (try to simplify variables name, but keep readability) which could return two action with fetched data and so on.

Yes, i see it's about time to implement redux in this project

> Btw, why have you used await this.checkDate() even if it's not async function?

I was messing around for some times with fetching necessary data and i guess i wrote it automatically without thinking what I'm actually doing :/

File: client/src/components/RepertoirePage.jsx (hunk 2 above)
Comment (BohdanOne) [reply client/src/components/RepertoirePage.jsx]: ok, there is more places in my code with bit more complex logic inside ternary operator so will track them and refactor too

**Pull request merged**